src/relations/mysql_provider.py

```diff
@@ -54,6 +56,10 @@ def __init__(self, charm) -> None:
         self.framework.observe(
             self.charm.on[DB_RELATION_NAME].relation_broken, self._on_database_broken
         )
+        self.framework.observe(
+            self.charm.on[DB_RELATION_NAME].relation_departed,
+            self._on_database_provides_relation_departed,
+        )
```

Comment:
maybe some comments good to add for better readability
```suggestion
        # cleanup users created by router for router<->app relation
        self.framework.observe(
            self.charm.on[DB_RELATION_NAME].relation_broken, self._on_database_broken
        )
        
        # cleanup router metadata and the router user
        self.framework.observe(
            self.charm.on[DB_RELATION_NAME].relation_departed,
            self._on_database_provides_relation_departed,
        )
```